web-admin/src/features/dashboards/DashboardAdminStateProvider.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+  import { page } from "$app/stores";
+  import { useQueryClient } from "@rilldata/svelte-query";
+  import { getBookmarks } from "@rilldata/web-admin/features/bookmarks/selectors";
+  import {
+    createTimeRangeSummary,
+    useMetricsView,
+    useModelHasTimeSeries,
+  } from "@rilldata/web-common/features/dashboards/selectors/index";
+  import { getStateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
+  import { metricsExplorerStore } from "@rilldata/web-common/features/dashboards/stores/dashboard-stores";
+  import { hasPersistentDashboardData } from "@rilldata/web-common/features/dashboards/stores/persistent-dashboard-state";
+  import { syncDashboardState } from "@rilldata/web-common/features/dashboards/stores/syncDashboardState";
+  import { initLocalUserPreferenceStore } from "@rilldata/web-common/features/dashboards/user-preferences";
+  import { createQueryServiceMetricsViewSchema } from "@rilldata/web-common/runtime-client";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import Spinner from "@rilldata/web-common/features/entity-management/Spinner.svelte";
+  import { EntityStatus } from "@rilldata/web-common/features/entity-management/types";
+
+  export let metricViewName: string;
+
+  $: initLocalUserPreferenceStore(metricViewName);
+  const stateManagers = getStateManagers();
+  const metricsView = useMetricsView(stateManagers);
+  const hasTimeSeries = useModelHasTimeSeries(stateManagers);
+  const timeRangeQuery = createTimeRangeSummary(stateManagers);
+  const metricsViewSchema = createQueryServiceMetricsViewSchema(
+    $runtime.instanceId,
+    metricViewName,
+  );
+
+  const queryClient = useQueryClient();
+  $: bookmarks = getBookmarks(
+    queryClient,
+    $runtime?.instanceId,
+    $page.params.organization,
+    $page.params.project,
+    metricViewName,
+  );
+
+  function syncDashboardStateLocal() {
```

Comment:
It's a bit confusing to see "Local" is this function name given this function is called in the Cloud context. I can't tell what "Local" refers to.